l10n_it_invoices_data_communication/models/communication.py

```diff
@@ -0,0 +1,2538 @@
+                    'dati_fattura_iva_ids':
+                        fattura._get_tax_comunicazione_dati_iva()
+                }
+                val = self._prepare_fattura_emessa(val, fattura)
+                vals_fatture.append((0, 0, val))
+
+            val_cessionario = {
+                'partner_id': cessionario.id,
+                'fatture_emesse_body_ids': vals_fatture
+            }
+            vals = self._prepare_cessionario_partner_id(
+                cessionario)
+            val_cessionario.update(vals)
+            dati_fatture.append((0, 0, val_cessionario))
+        return dati_fatture
+
+    def compute_fatture_emesse(self):
+        self.ensure_one()
+        fatture_emesse = self._get_fatture_emesse()
+        if fatture_emesse:
+            # Cedente
+            self.partner_cedente_id = \
+                fatture_emesse[0].company_id.partner_id.id
+            self.onchange_partner_cedente_id()
+
+            # Cessionari
+            cessionari = fatture_emesse.mapped('partner_id')
+            dati_fatture = self._prepare_cessionari_dati_fatture(
+                fatture_emesse, cessionari)
+            self.fatture_emesse_ids = dati_fatture
+
+    def _get_fatture_emesse_domain(self):
+        domain = [('comunicazione_dati_iva_escludi', '=', True)]
+        no_journal_ids = self.env['account.journal'].search(domain).ids
+        domain = [('type', 'in', ['out_invoice', 'out_refund']),
+                  ('comunicazione_dati_iva_escludi', '=', False),
+                  ('move_id', '!=', False),
+                  ('move_id.journal_id', 'not in', no_journal_ids),
+                  ('company_id', '=', self.company_id.id),
+                  ('date_invoice', '>=', self.date_start),
+                  ('date_invoice', '<=', self.date_end),
+                  '|',
+                  ('fiscal_document_type_id.out_invoice', '=', True),
+                  ('fiscal_document_type_id.out_refund', '=', True),
+                  ]
+        return domain
+
+    def _get_fatture_emesse(self):
+        self.ensure_one()
+        domain = self._get_fatture_emesse_domain()
+        return self.env['account.invoice'].search(domain)
+
+    def _prepare_cedenti_dati_fatture(self, fatture_ricevute, cedenti):
+        dati_fatture = []
+        posizione = 0
+        for cedente in cedenti:
+            # Fatture
+            fatture = fatture_ricevute.filtered(
+                lambda fr: fr.partner_id.id == cedente.id)
+            vals_fatture = []
+            for fattura in fatture:
+                posizione += 1
+                val = {
+                    'posizione': posizione,
+                    'invoice_id': fattura.id,
+                    'dati_fattura_TipoDocumento':
+                        fattura.fiscal_document_type_id.id,
+                    'dati_fattura_Data': fattura.date_invoice,
+                    'dati_fattura_DataRegistrazione':
+                        fattura.date_invoice,
```

Comment:
perchè `date_invoice` qui? per capire, sei ripartito da zero o dalla PR #1175?

Reply:
@sergiocorato sono ripartito da 0 con il backporting di #1005.
In #1175 non capivo cosa avevi modificato e cosa invece era in v10.

Reply:
@SimoRubi 
Suggerisco allora una delle 2 strade:

 - Partire da https://github.com/openerp-italia/modules/tree/8.0/l10n_it_comunicazione_dati_iva (in produzione da tempo su diversi clienti; per chi non ha accesso: scusate il repo privato) e aggiungere le funzionalità aggiunte su 10.0 con https://github.com/OCA/l10n-italy/pull/1005
 - Partire da https://github.com/OCA/l10n-italy/pull/1175 e verificare cosa manca

Reply:
@eLBati credo che @SimoRubi sia partito da #1005 (anch'io) per cui possiamo proseguire qui senza problemi, verifico cosa avevo aggiunto (casomai @SimoRubi vedi se trovi qualche altre correzione sul repo privato)

Reply:
@eLBati applicare le tue modifiche a https://github.com/openerp-italia/modules/tree/8.0/l10n_it_comunicazione_dati_iva è tutto un grosso conflitto.
Procedo a fare il confronto manuale, con l'aiuto delle review di @sergiocorato

Reply:
con @eLBati abbiamo controllato e la prima strada che suggeriva non è percorribile per via di troppi conflitti

Reply:
@SimoRubi la domanda da cui era partita la conversazione è rimasta irrisolta, perchè `date_invoice` nel campo `dati_fattura_DataRegistrazione` ? il campo `date_invoice` è già usato sopra

Reply:
È così anche in v10, comunque hai ragione, ho dimenticato di applicare quanto dico in https://github.com/OCA/l10n-italy/pull/1211/files#r286948924 anche qui.
Appena posso lo faccio, grazie

Reply:
Fatto